scaleway/validation.go

```diff
@@ -0,0 +1,68 @@
+package scaleway
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+// validationUUID validate the schema follows the canonical UUID format
+// "6ba7b810-9dad-11d1-80b4-00c04fd430c8".
+func validationUUID() func(interface{}, string) ([]string, []error) {
+	return func(v interface{}, key string) (warnings []string, errors []error) {
+		uuid, isString := v.(string)
+		if !isString {
+			return nil, []error{fmt.Errorf("invalid UUID: not a string")}
+		}
+
+		t := []byte(uuid)
+		if len(t) != 36 || t[8] != '-' || t[13] != '-' || t[18] != '-' || t[23] != '-' {
```

Comment:
Wouldn't it better to use a uuid parsing lib here?

Reply:
Didn't want to add a dep for one function.